frontend/src/AppBuilder/CodeEditor/PreviewBox.jsx

```diff
@@ -96,7 +96,7 @@ export const PreviewBox = ({
   const [largeDataset, setLargeDataset] = useState(false);
   const globals = useStore((state) => state.getAllExposedValues().constants || {}, shallow);
   const secrets = useStore((state) => state.getSecrets(), shallow);
-  const globalServerConstantsRegex = /^\{\{.*globals\.server.*\}\}$/;
+  const globalServerConstantsRegex = /\{\{.*globals\.server.*\}\}/;
```

Comment:
SonarCloud test must be failing because of the regex. You can try 
`const globalServerConstantsRegex = /\{\{[^}]*globals\.server[^}]*\}\}/;`  (this will eliminate the super-linear backtracking)